pkg/master/controller.go

```diff
@@ -142,12 +143,19 @@ func (c *Controller) CreateNamespaceIfNeeded(ns string) error {
 
 // CreateMasterServiceIfNeeded will create the specified service if it
 // doesn't already exist.
-func (c *Controller) CreateMasterServiceIfNeeded(serviceName string, serviceIP net.IP, servicePort int) error {
+func (c *Controller) CreateMasterServiceIfNeeded(serviceName string, serviceIP net.IP, servicePort int, serviceEnableDNS bool) error {
 	ctx := api.NewDefaultContext()
 	if _, err := c.ServiceRegistry.GetService(ctx, serviceName); err == nil {
 		// The service already exists.
 		return nil
 	}
+	servicePorts := []api.ServicePort{{Name: "https", Port: servicePort, Protocol: api.ProtocolTCP, TargetPort: util.NewIntOrStringFromInt(servicePort)}}
+	if serviceEnableDNS {
+		servicePorts = append(servicePorts,
+			api.ServicePort{Name: "dns-tcp", Port: 53, Protocol: api.ProtocolTCP, TargetPort: util.NewIntOrStringFromInt(53)},
```

Comment:
I don't understand - who is actually serving this DNS?  Did someone add a DNS server to the master while I wasn't looking?